libafl/src/executors/hooks/inprocess.rs

```diff
@@ -196,9 +196,22 @@ impl<I, S> ExecutorHook<I, S> for InProcessHooks<I, S> {
     fn init(&mut self, _state: &mut S) {}
     /// Call before running a target.
     fn pre_exec(&mut self, _state: &mut S, _input: &I) {
+        // # Safety
+        // _Technically_ this and `post_exec` could race against each other.
+        // Theoretically, one could end up in a state with timeout_handler set, and crash_handler not,
+        // or vice versa.
+        // In practice though, it's not clear how this would be an issue.
+        // Therefore, we will ignore this very slight edge-case for performance.
+        // We are a fuzzer, after all.
         #[cfg(feature = "std")]
         unsafe {
             let data = &raw mut GLOBAL_STATE;
+
+            assert!(
+                ((*data).timeout_handler.is_null()) && ((*data).crash_handler.is_null()),
+                "Called InProcesssHooks::pre_exec multiple times! This is not supported."
+            );
```

Comment:
these two pointers should be null at this time no?

Reply:
Yes they should, hence the assert

Reply:
ah ok
but
this whole InProcessExecutorHandlerData is not gonna work if it is multithreadded.
but things will break even before this assert is even reached so it's useless